packages/editor/src/client/atoms/dateTimePicker.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import {DatePicker, FormGroup, ControlLabel, Form} from 'rsuite'
+import {useTranslation} from 'react-i18next'
+
+export interface DateTimeProps {
+  dateTime?: Date
+  label: string
+  changeDate(publishDate?: Date): void
+}
+
+export function DateTimePicker({dateTime, label, changeDate}: DateTimeProps) {
+  const {t} = useTranslation()
+
+  return (
+    <Form fluid={true}>
+      <FormGroup>
+        <ControlLabel>{label}</ControlLabel>
+        <DatePicker
+          style={{marginRight: 8}}
+          placement="auto"
+          value={dateTime?.getTime() === dateTime?.getTime() ? dateTime : undefined}
+          format="DD MMM YYYY"
+          ranges={[
+            {
+              label: t('timeDatePicker.today'),
+              value: new Date()
+            },
+            {
+              label: t('timeDatePicker.tomorrow'),
+              value: () => {
+                const tomorrow = new Date()
+                tomorrow.setDate(new Date().getDate() + 1)
+                return tomorrow
+              }
+            },
+            {
+              label: t('timeDatePicker.nextSaturday'),
+              value: () => {
+                const nextSaturday = new Date()
```

Comment:
i'm not sure if this works. Maybe this article helps: https://codereview.stackexchange.com/questions/33527/find-next-occurring-friday-or-any-dayofweek

Reply:
`i` is remaining days till Saturday. I add them to the current date, unless it's already Saturday and add 7 instead. I think that logic works? I tested with a few different dates in the console, and looks right. I think the solution you linked is essentially the same thing, but with any day besides Saturday you need to handle a negative result for remaining days  so add the `+7` and `%`.